code/game/objects/items/weapons/RCL.dm

```diff
@@ -0,0 +1,205 @@
+#define RCLMAX 30
+
+/obj/item/weapon/rcl
+	name = "rapid cable layer (RCL)"
+	desc = "A device used to rapidly deploy cables."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "rcl-0"
+	item_state = "rcl-0"
+	opacity = 0
+	flags = FPRINT
+	siemens_coefficient = 1 //Not quite as conductive as working with cables themselves
+	force = 5.0 //Plastic is soft
+	throwforce = 5.0
+	throw_speed = 1
+	throw_range = 10
+	w_class = 3.0
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = "engineering=2;materials=4"
+	//var/active = 0 Depreciated; Leaving it because it is a useful framework tool if you want to make it automagically place on movement
+	var/obj/structure/cable/last = null
+	var/cables = 0
+
+/obj/item/weapon/rcl/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W,/obj/item/stack/cable_coil))
+		var/obj/item/stack/cable_coil/C = W
+		var/calc = min(RCLMAX - cables,C.amount)
+		cables += calc
+		update_icon()
+		C.use(calc)
+		user << "<span class='notice'>You add the cables to the RCL. It now contains [cables].</span>"
+	else
+		..()
+
+/obj/item/weapon/rcl/examine(mob/user)
+	..()
+	user << "<span class='info'>It contains [cables]/30 cables.</span>"

```

Comment:
You should add a max cables variable for adminbus, instead of hardcoding it at 30